provider/lxd/credentials.go

```diff
@@ -368,14 +355,14 @@ func (p environProviderCredentials) finalizeCredential(
 		return nil, errors.NotValidf("missing or empty %q attribute", credAttrClientKey)
 	}
 
-	isLocalEndpoint, err := p.isLocalEndpoint(args.CloudEndpoint)
```

Comment:
I think we should probably just guard here with `IsDefaultCloud`. This will contact the server over the socket.